pkg/commands/copy_test.go

```diff
@@ -66,6 +66,11 @@ var copyTests = []struct {
 		sourcesAndDest: []string{"f[o][osp]", "tempCopyExecuteTest"},
 		expectedDest:   []string{"tempCopyExecuteTest"},
 	},
+	{
+		name:           "Copy into several to-be-created directories",
+		sourcesAndDest: []string{"f[o][osp]", "tempCopyExecuteTest/foo/bar"},
```

Comment:
I understand that you included in the PR message about adding the test but it looks like this is not adding coverage for the `CreateFile` case with UID/GID :/

Reply:
What might be valuable here is to add an integration test at https://github.com/GoogleContainerTools/kaniko/tree/main/integration/dockerfiles, with `COPY --chown= ...` and it would be very helpful so that we could verify the fix when comparing in the future release.

Reply:
Yeah, you're right, it doesn't set UID/GID. I didn't know how to add that test :( I think I added this one to quickly be able to figure out how to create intermediate directories in the first place.

I'll add an integration test.